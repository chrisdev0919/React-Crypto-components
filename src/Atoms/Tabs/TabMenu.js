import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const BaseContainer = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  align-items: center;
  width: 100%;
  border-bottom: 2px solid ${props => props.theme['gray-1']};
  box-sizing: border-box;
`

const BaseItem = styled.div`
  position: relative;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  width: auto;
  padding: 15px;
  box-sizing: border-box;
  cursor: pointer;

  &:hover { & :last-child { display: inline-block!important; } }
`
const ItemBar = styled.div`
  display: ${props => props.visible ? 'inline-block' : 'none'}; 
  position: absolute;
  bottom: -2px;
  left: 0;
  width: 100%;
  height: 5px;
  background-color: ${props => props.theme['gold']};
`

const Container = ({ children }) => (
  <BaseContainer>
    {children}
  </BaseContainer>
)

const Item = ({ children, selected, rest }) => (
  <BaseItem {...rest}>
    {children}
    <ItemBar visible={selected} />
  </BaseItem>
)

Item.propTypes = {
  selected: PropTypes.bool
}

Item.defaultProps = {
  selected: false
}

export {
  Container,
  Item
}


