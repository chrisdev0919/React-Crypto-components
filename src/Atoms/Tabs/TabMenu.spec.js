import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import TabMenu from './TabMenu'

describe('TabMenu component', () => {
  it('renders correctly', () => {
    const component = shallow(<TabMenu>MyTabMenu</TabMenu>)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
