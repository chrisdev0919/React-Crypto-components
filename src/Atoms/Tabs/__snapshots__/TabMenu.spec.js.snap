// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabMenu component renders correctly 1`] = `
<styled.div>
  MyTabMenu
</styled.div>
`;
